vid/src/avid_m/proofs.rs

```diff
@@ -0,0 +1,224 @@
+//! This module implements encoding proofs for the Avid-M Scheme.
+
+use std::collections::HashSet;
+
+use jf_merkle_tree::MerkleTreeScheme;
+use jf_utils::canonical;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    avid_m::{
+        config::AvidMConfig, AvidMCommit, AvidMParam, AvidMScheme, AvidMShare, Config, MerkleProof,
+        MerkleTree, F,
+    },
+    VerificationResult, VidError, VidResult, VidScheme,
+};
+
+/// A proof of incorrect encoding.
+/// It consists of a witness vector, which is the claimed low degree polynomial,
+/// and a list of [`MalEncodingProofRawShare`]s. Each raw share claims that, in the Merkle tree committed by the provided commitment,
+/// some code word of the witness vector appears at the `index` position.
+/// If we have enough raw shares, and the Merkle commitment of the encoding of witness vector doesn't match the provided commitment,
+/// we assert that the provided commitment cannot open to be a correct encoding.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct MalEncodingProof {
+    /// The witness vector
+    #[serde(with = "canonical")]
+    witness: Vec<F>,
+    /// Proof content.
+    raw_shares: Vec<MalEncodingProofRawShare>,
+}
+
+/// Proof content, see [`MalEncodingProof`].
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct MalEncodingProofRawShare {
+    /// The index of this raw share
+    pub index: usize,
+    /// A Merkle tree proof against the provided Merkle tree commitment.
+    #[serde(with = "canonical")]
+    pub mt_proof: MerkleProof,
+}
```

Comment:
what about: 
- rename `witness` as `interpolated_poly` or `recovered_poly` or `recovered_data`? 
- remove the `MalEncodingProofRawShare` struct, as seems to be unnecessary abstraction, instead, simply `raw_shares: Vec<(usize, MerkleProof)>`, with comment explaining the tuple.
- avoid language like "proof content" this sounds like `raw_shares` is the actual proof, which it isn't. The whole bundle serve as a proof.

Reply:
Done also in [8e6a8d6](https://github.com/EspressoSystems/espresso-network/pull/2803/commits/8e6a8d63a2972ff020999dc8f9cc84381ac03cf8)